sympy/assumptions/refine.py

```diff
@@ -336,6 +337,26 @@ def refine_sign(expr, assumptions):
     return expr
 
 
+def refine_matrixelement(expr, assumptions):
+    """
+    Examples
+    ========
+        >>>refine(X[0, 1], Q.symmetric(X))
+        X[0, 1]
+        >>> refine(X[1, 0], Q.symmetric(X))
+        X[0,1]
+    """ 
+    arg = expr.args[0]
+    if ask(Q.symmetric(arg), assumptions):
+        if (expr.args[1] > expr.args[2]):
```

Comment:
What about `if ask(Q.symmetric(arg), assumptions) and (expr.args[1] > expr.args[2]):`?
Is it possible for `expr.args[1]` or `expr.args[2]` to be symbolic? How it works in that case?

Reply:
`refine_matrixelement(X[0, 1], Q.symmetric(X))`.
here expr.args[1] produces 0 and expr.args[2] produces 1.  and expected result should be `x[0,1]`